segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/exts3/EXTS3StorageTest.java

```diff
@@ -0,0 +1,279 @@
+import org.jclouds.ContextBuilder;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.channels.FileChannel;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+/**
+ * Unit tests for ExtS3Storage.
+ */
+@Slf4j
+public class EXTS3StorageTest extends IdempotentStorageTest {
+    private ExtS3StorageFactory storageFactory;
+    private ExtS3StorageConfig adapterConfig;
+    private S3JerseyClient client = null;
+    private S3Proxy s3Proxy;
+
+    @Before
+    public void setUp() throws Exception {
+
+        String endpoint = "http://127.0.0.1:9020";
+        URI uri = URI.create(endpoint);
+        Properties properties = new Properties();
+        properties.setProperty("s3proxy.authorization", "none");
+        properties.setProperty("s3proxy.endpoint", endpoint);
+        properties.setProperty("jclouds.provider", "filesystem");
+        properties.setProperty("jclouds.filesystem.basedir", "/tmp/s3proxy");
+
+        ContextBuilder builder = ContextBuilder
+                .newBuilder("filesystem")
+                .credentials("x", "x")
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .overrides(properties);
+        BlobStoreContext context = builder.build(BlobStoreContext.class);
+        BlobStore blobStore = context.getBlobStore();
+
+        s3Proxy = S3Proxy.builder().awsAuthentication( AuthenticationType.AWS_V2_OR_V4, "x", "x")
+                .endpoint(uri)
+                .keyStore("", "")
+                .blobStore(blobStore)
+                .ignoreUnknownHeaders(true)
+                .build();
+
+        s3Proxy.start();
+
+        this.adapterConfig = ExtS3StorageConfig.builder()
+                                               .with(ExtS3StorageConfig.EXTS3_BUCKET, "kanpravegatest")
+                                               .with(ExtS3StorageConfig.EXTS3_ACCESS_KEY_ID, "x")
+                                               .with(ExtS3StorageConfig.EXTS3_SECRET_KEY, "x")
+                                               .with(ExtS3StorageConfig.ROOT, "test")
+                                               .with(ExtS3StorageConfig.EXTS3_URI, "http://127.0.0.1:9020")
+                                               .build();
+        if (client == null) {
+            try {
+                createStorage();
+
+                try {
+                    client.createBucket("kanpravegatest");
+                } catch (Exception e) {
+                    if (!( e instanceof S3Exception && ((S3Exception) e).getErrorCode().
```

Comment:
This was throwing all the exceptions except `BucketAlreadyOwnedByYou` changed the code to make it simpler.